### PR TITLE
Support running pipelines on main in .jenkins and tools

### LINKED ISSUE #71806: Migrate master to main: https://github.com/pytorch/pytorch

### PR DESCRIPTION
**Pull request by musebc**
## Release Notes
This commit updates the tools and .jenkins folders to support running on both `master` and `main` branches in preparation for the transition.

## Topics
Code cleanup

Fixes #71806

### CODE REVIEW

**Comment by suo** [comment]
driveby comment: this seems like the rare case where a global env var is the best solution. If we just set $GIT_DEFAULT_BRANCH at the beginning of everything and use that throughout the codebase (and eventually delete it in favor of direct references to main) it's probably cleaner than routing arguments through all the scripts.

File: .jenkins/pytorch/short-perf-test-cpu.sh [hunk 1]
```diff
@@ -18,25 +18,26 @@ pip install -q awscli
 # More info at https://github.com/aws/aws-cli/issues/2321
 aws configure set default.s3.multipart_threshold 5GB
 
-if [[ "$COMMIT_SOURCE" == master ]]; then
-    # Get current master commit hash
+if [[ "$COMMIT_SOURCE" == master || "$COMMIT_SOURCE" == main ]]; then
+    # Get current main commit hash
     MASTER_COMMIT_ID=$(git log --format="%H" -n 1)
```
Comment (janeyx99) [inline .jenkins/pytorch/short-perf-test-cpu.sh:23]: do we want to rename these vars as well?

File: .jenkins/pytorch/short-perf-test-cpu.sh [hunk 2]
```diff
@@ -18,25 +18,26 @@ pip install -q awscli
...
     if aws s3 ls s3://ossci-perf-test/pytorch/cpu_runtime/"${commit_id}".json; then
         LATEST_TESTED_COMMIT=${commit_id}
         break
     fi
-done < <(git rev-list upstream/master)
+done < <(git rev-list upstream/$UPSTREAM_DEFAULT_BRANCH)
 aws s3 cp s3://ossci-perf-test/pytorch/cpu_runtime/"${LATEST_TESTED_COMMIT}".json cpu_runtime.json
 
-if [[ "$COMMIT_SOURCE" == master ]]; then
+if [[ "$COMMIT_SOURCE" == master || "$COMMIT_SOURCE" == main ]]; then
```
Comment (janeyx99) [inline .jenkins/pytorch/short-perf-test-cpu.sh:40]:
```suggestion
if [[ "$COMMIT_SOURCE" == "$UPSTREAM_DEFAULT_BRANCH" ]]; then
```
why not

Comment (janeyx99) [reply .jenkins/pytorch/short-perf-test-cpu.sh:40]: same for two other instances below

**Review by janeyx99 — CHANGES_REQUESTED** [review]
CI failures look legit

**Comment by musebc** [comment]
> CI failures look legit

Yep. taking a look now.

File: .jenkins/pytorch/short-perf-test-cpu.sh (hunk 2 above)
Comment (musebc) [reply .jenkins/pytorch/short-perf-test-cpu.sh]: Updated.

File: .jenkins/pytorch/short-perf-test-gpu.sh [hunk 3]
```diff
@@ -18,25 +18,26 @@ pip install -q awscli --ignore-installed PyYAML
...
     if aws s3 ls s3://ossci-perf-test/pytorch/gpu_runtime/"${commit_id}".json; then
         LATEST_TESTED_COMMIT=${commit_id}
         break
     fi
-done < <(git rev-list upstream/master)
+done < <(git rev-list upstream/$UPSTREAM_DEFAULT_BRANCH)
 aws s3 cp s3://ossci-perf-test/pytorch/gpu_runtime/"${LATEST_TESTED_COMMIT}".json gpu_runtime.json
 
-if [[ "$COMMIT_SOURCE" == master ]]; then
+if [[ "$COMMIT_SOURCE" == master || "$COMMIT_SOURCE" == main ]]; then
```
[the remark this answers is not in the document]
Comment (musebc) [reply .jenkins/pytorch/short-perf-test-gpu.sh]: Yep. Updated.

File: .jenkins/pytorch/short-perf-test-gpu.sh [hunk 4]
```diff
@@ -17,14 +17,15 @@ pip install -q awscli --ignore-installed PyYAML
...
 # More info at https://github.com/aws/aws-cli/issues/2321
 aws configure set default.s3.multipart_threshold 5GB
+UPSTREAM_DEFAULT_BRANCH="$(git remote show https://github.com/pytorch/pytorch.git | awk '/HEAD branch/ {print $NF}')"
 
-if [[ "$COMMIT_SOURCE" == master ]]; then
-    # Get current master commit hash
-    MASTER_COMMIT_ID=$(git log --format="%H" -n 1)
-    export MASTER_COMMIT_ID
+if [[ "$COMMIT_SOURCE" == "$UPSTREAM_DEFAULT_BRANCH" ]]; then
+    # Get current main commit hash
```
Comment (malfet) [inline .jenkins/pytorch/short-perf-test-gpu.sh:23]: s/main/default branch/
```suggestion
    # Get current default branch commit hash
```

**Review by malfet — APPROVED** [review]
LGTM, but please avoid "main" in the comments, use "default branch"/"trunk"

**Pull request closed without merging**